Makefile

```
@@ -14,11 +14,12 @@ endif
 OUT := out$(SUFFIX)
 
 define safe_flag
-$(shell $(CC) $(if $(filter clang,$(CC)),-Werror=unknown-warning-option) -E $1 - </dev/null >/dev/null 2>&1 && echo $1 || echo $2)
+$(shell $(CC) $(if $(filter clang%,$(CC)),-Werror=unknown-warning-option) -E $1 - </dev/null >/dev/null 2>&1 && echo $1 || echo $2)
 endef
 
 CPPFLAGS := $(CPPFLAGS) -D_GNU_SOURCE -I include
-SHARED_FLAGS := -pipe -O3 -flto -fPIC -fvisibility=hidden -fno-plt \
+SHARED_FLAGS := -pipe -O3 -fPIC -fno-plt \
+    $(call safe_flag,-fsanitize=cfi -flto -fvisibility=hidden,-flto -fvisibility=hidden) \
```

`-flto` and `-fvisibility` are pretty standard in GCC and Clang, can this be simplified to just:

```suggestion
    $(call safe_flag,-fsanitize=cfi) -flto -fvisibility=hidden \
```

---

If `-fvisibility=hidden` and `-flto` are not used in the `safe_flag` test, the test will always fail since they are requirements for CFI, see https://clang.llvm.org/docs/ControlFlowIntegrity.html

---

This is the exact same thing, just simplified. I'm not removing LTO or visibiltiy.